frontend/src/app/hierarchy-viewer/hierarchy-viewer.component.ts

```diff
@@ -19,32 +23,194 @@ export class HierarchyViewerComponent implements OnDestroy {
-        //This will go away eventually. We will replace it with creating hierarchies
-        //from scratch
-        this.store.dispatch(createHierarchy({hierarchy: RRRHierarchy}));
-        //this.store.dispatch(createHierarchy({hierarchy: SimpleHierarchy}))
+        this.store.dispatch(createHierarchy({ hierarchy: RRRHierarchy }));
+    }
+
+    deleteHierarchy() {
+        if (this.selectedHierarchy) {
+            this.store.dispatch(deleteHierarchy({ hierarchyId: this.selectedHierarchy.id }));
+        }
     }
 
     onSelect(event: MatSelectChange) {
         this.selectedHierarchy = null;
-        this.store.dispatch(setSelectedHierarchy({selectedHierarchyId: event.value}));
+        this.store.dispatch(setSelectedHierarchy({ selectedHierarchyId: event.value }));
+    }
+
+    createHierarchyDialog() {
+        const dialogRef = this.dialog.open(CreateHierarchyDialog, {
+            data: { name: "", description: "" }
+        });
+
+        dialogRef.afterClosed().subscribe(result => {
+            if (result != null && result.event == 'Create') {
+                this.selectedHierarchy = null;
+                let hierarchyRequest: HierarchyRequest = {
+                    name: result.name, description: result.description, nodes: [], alternatives: []
+                };
+                this.store.dispatch(createHierarchy({ hierarchy: hierarchyRequest }));
+            }
+        });
+    }
+
+    deleteHierarchyDialog() {
+
+        const dialogRef = this.dialog.open(DeleteHierarchyDialog, {
+            data: { name: this.selectedHierarchy ? this.selectedHierarchy.name : "" }
+        });
+
+        dialogRef.afterClosed().subscribe(result => {
+
+            if (result != null && result.event == 'Delete') {
+                this.deleteHierarchy();
+                this.selectedHierarchy = null;
+            }
+        });
+    }
+
+    importExportHierarchyDialog() {
+        const dialogRef = this.dialog.open(ImportExportHierarchyDialog, {
+            data: { name: "", description: "", hierarchy: this.selectedHierarchy }
+        });
+
+        dialogRef.afterClosed().subscribe(result => {
+            if (result != null && result.event == 'Import') {
+                const importFile: File = result.file;
+
+                const fileReader = new FileReader();
+                fileReader.readAsText(importFile, "UTF-8");
+                fileReader.onload = () => {
+                    const hierarchyRequest: HierarchyRequest = JSON.parse(fileReader.result as string);
+                    this.selectedHierarchy = null;
+
+                    //The exported file contains ids and icons, so we need to fix that on import or export
+                    this.store.dispatch(createHierarchy({ hierarchy: hierarchyRequest }));
+                }
+                fileReader.onerror = (error) => {
+                    console.log(error);
+                }
+            }
+        });
+    }
+}
+
+@Component({
```

Comment:
Break the create and delete Hierarchy dialogs into their own separate files. Group each into its own folder inside of the hierarchy-viewer folder.